pkg/registry/core/service/storage/rest.go

```diff
@@ -233,7 +233,7 @@ func (rs *REST) Delete(ctx context.Context, id string, options *metav1.DeleteOpt
 	if !dryrun.IsDryRun(options.DryRun) {
 		// TODO: can leave dangling endpoints, and potentially return incorrect
 		// endpoints if a new service is created with the same name
-		_, _, err = rs.endpoints.Delete(ctx, id, &metav1.DeleteOptions{})
+		_, _, err = rs.endpoints.Delete(ctx, id, rest.ValidateAllObjectFunc, &metav1.DeleteOptions{})
```

Comment:
interesting... does this mean delete admission is not (and was not) done on endpoints deleted by the service REST handler?

Reply:
Right. I think it's ok. If needed, the service webhook can check the endpoints when validating the service deletion.